pkgs/development/libraries/science/math/magma/generic.nix

```diff
@@ -25,7 +20,9 @@
 , magmaRelease
 , ninja
 , config
-, rocmSupport ? config.rocmSupport
+  # At least one back-end has to be enabled,
+  # and we can't default to CUDA since it's unfree
+, rocmSupport ? !cudaSupport
```

Comment:
This diff strikes me as suspect... IIUC this package builds iff exactly one of (cudaSupport, rocmSupport) is enabled. In the spirit of [making illegal states unrepresentable](https://blog.janestreet.com/effective-ml-revisited/), it seems to me that we should support only a single flag instead of two boolean flags. 

Admittedly, I do not have an obvious choice of name or design for such a flag of the top of my head. Perhaps
```
, backend ? "cuda"
```
where `backend` serves as a sort of enum for the two backend options?

Reply:
Perhaps something like what's seen here: 9709e96e1e62b925e9f68809040913169fe07d74?

Reply:
https://github.com/NixOS/nixpkgs/blob/9709e96e1e62b925e9f68809040913169fe07d74/pkgs/applications/science/chemistry/cp2k/default.nix#L50
Also note the above.

Reply:
> Perhaps something like what's seen here: https://github.com/NixOS/nixpkgs/commit/9709e96e1e62b925e9f68809040913169fe07d74?

yes, exactly

Reply:
We could do that, or we could keep the existing interface and deprecate it properly in a separate PR. If one attempts to use an invalid combination they stumble on `broken = true`

Reply:
I think the linked issue is a better idea, and we can include deprecation notices that way.

Reply:
@samuela is this sound?

Reply:
Yup, sounds good to me 👍